### PR TITLE
Add Make manifests command to generate install.yaml

### LINKED ISSUE #121: Makefile Target to auto-generate install.yaml
# Summary

We need a target similar to Numaflow's "make manifests" to generate an [install.yaml file](https://github.com/numaproj/numaflow/blob/v1.1.6/Makefile#L210) to persist in our repository. (once we have Github version tags, we can grab the correct file pertaining to the version)

One difference from Numaflow is that our file will not be a complete deployable file on its own. This is because we require the [cluster name to be included in our ConfigMap](https://github.com/numaproj-labs/numaplane/blob/b2468d51bcab82dfef8d918a15f5d031cf40c708/config/manager/config.yaml#L8). The idea is that the organization using this manifest is expected to set this themselves as part of their kustomization built on top of this file.

We do still need to be able to do tests locally as part of "make start", so we will need to set the field in the ConfigMap as part of that (as we do [here](https://github.com/numaproj-labs/numaplane/blob/95066f30e304d343a592fa83ba71ab2279669d1b/Makefile#L157)). We should remove that field from our main ConfigMap spec, and instead separate it into a ConfigMap Patch file which only gets run when testing locally.

This Makefile target should be called from (or a dependency of) "make codegen" - which is an Issue defined [here](https://github.com/numaproj-labs/numaplane/issues/78).

# Use Cases

When would you use this?

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by chandankumar4**
Fixes #121 

### Modifications

Add "make manifests" command to generate `install.yaml`

### Verification

Verified by applying install.yaml in local k8s cluster

### CODE REVIEW

File: config/manager/config.yaml [hunk 1]
```diff
@@ -6,7 +6,6 @@ metadata:
   namespace: numaplane-system
 data:
   config.yaml: |
-    clusterName: "CLUSTER_NAME_VALUE" # Replace with the cluster name where numaplane will get deployed.
```
Comment (chandankumar4) [inline config/manager/config.yaml:9]: we are using a custom config for `make start`, so we don't need to patch anything for local

File: config/manager/config.yaml [hunk 2]
```diff
@@ -6,7 +6,6 @@ metadata:
...
     timeIntervalSec: 60
     repoCredentials:
     - url: "github.com/numaproj-labs"
```
Comment (chandankumar4) [inline config/manager/config.yaml:11]: Should we remove this complete data from config map? as this is for testing purpose and not for prod and user should add these value before deploying

Comment (juliev0) [reply config/manager/config.yaml:11]: good point, yes, the repoCredentials shouldn't be in there either. I kind of think we can leave the `timeIntervalSec` in there and just consider that to be the default.

Comment (chandankumar4) [reply config/manager/config.yaml]: okay, Made the changes accordingly

**Review by juliev0 — APPROVED** [review]

**Review by xdevxy — APPROVED** [review]

**Pull request merged**